internal/middleware/compression.go

```diff
@@ -20,25 +20,59 @@ import (
 	"compress/zlib"
 	"io"
 	"net/http"
+
+	"github.com/golang/snappy"
+	"github.com/klauspost/compress/zstd"
 )
 
 const (
 	headerContentEncoding = "Content-Encoding"
-	headerValueGZIP       = "gzip"
+	CompressionGzip       = "gzip"
+	CompressionZlib       = "zlib"
+	CompressionDeflate    = "deflate"
+	CompressionSnappy     = "snappy"
+	CompressionZstd       = "zstd"
```

Comment:
Can these values be shared between gRPC and HTTP? I remember to see them there as well.

Reply:
It can be shareable. I removed usages of these values in confighttp and made connection to middleware.CompressionType.
To connect these values with configgrpc, I discussed with @Aneurysm9 and since the changes in confighttp are quite a lot, he will create an issue about refactoring configgrpc after this PR is being merged.